app/pages/clinicworkspace/ClinicPatients.js

```diff
@@ -724,6 +725,9 @@ export const ClinicPatients = (props) => {
   }, [deletingClinicPatientTag, handleAsyncResult, handleCloseClinicPatientTagUpdateDialog, previousDeletingClinicPatientTag?.inProgress, t]);
 
   useEffect(() => {
+    // Prevent this effect from firing on logout, which would clear all activeFilters from localStorage
+    if (!clinic) return;
+
```

Comment:
The patientTags are actually being stored properly in localStorage. However, they are being wiped out by the firing of this function on user logout. 

This function's responsibility is to remove tags from the currently applied set of filters if they have been removed from the clinic's catalogue of tags. When the clinic's tag library is modified, the effect is fired, and the function un-applies all tag filters that don't exist in the clinic tag library anymore.

The issue is that when the user logs out, the clinic is momentarily set to `undefined`. This component sees that the clinic has no tags and assumes ALL tags have been removed from the clinic library, and therefore we should clear all filters.

Reply:
Good job diagnosing this :+1: